packages/framework/tests/Unit/Component/Money/MoneyTest.php

```diff
@@ -0,0 +1,582 @@
+    }
+
+    public function testSubtractIsImmutable(): void
+    {
+        $money = Money::fromString('1');
+
+        $money->subtract(Money::fromString('1'));
+
+        $this->assertEquals('1', $money->toString());
+    }
+
+    /**
+     * @dataProvider subtractProvider
+     * @param string $a
+     * @param string $b
+     * @param string $expectedResult
+     */
+    public function testSubtract(string $a, string $b, string $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+        $moneyB = Money::fromString($b);
+
+        $result = $moneyA->subtract($moneyB);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function subtractProvider(): Iterator
+    {
+        yield ['2', '1', '1'];
+        yield ['12.15', '34.965', '-22.815'];
+        yield ['10', '-2', '12'];
+        yield ['1', '0.01', '0.99'];
+        yield ['0.5', '0.5', '0.0'];
+        yield ['1.525', '0.475', '1.050'];
+        yield ['1.00', '1.000', '0.000'];
+    }
+
+    public function testMultiplyIsImmutable(): void
+    {
+        $money = Money::fromString('1');
+
+        $money->multiply('2');
+
+        $this->assertEquals('1', $money->toString());
+    }
+
+    /**
+     * @dataProvider multiplyProvider
+     * @param string $a
+     * @param string $b
+     * @param string $expectedResult
+     */
+    public function testMultiply(string $a, string $b, string $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+
+        $result = $moneyA->multiply($b);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function multiplyProvider(): Iterator
+    {
+        yield ['2', '1', '2'];
+        yield ['12.15', '34.965', '424.82475'];
+        yield ['10', '-2', '-20'];
+        yield ['1', '0.01', '0.01'];
+        yield ['0.5', '0.5', '0.25'];
+        yield ['1.525', '0.475', '0.724375'];
+        yield ['1.00', '1.000', '1.00000'];
+    }
+
+    public function testDivideIsImmutable(): void
+    {
+        $money = Money::fromString('1');
+
+        $money->divide('2', 1);
+
+        $this->assertEquals('1', $money->toString());
+    }
+
+    /**
+     * @dataProvider divideProvider
+     * @param string $a
+     * @param string $b
+     * @param int $scale
+     * @param string $expectedResult
+     */
+    public function testDivide(string $a, string $b, int $scale, string $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+
+        $result = $moneyA->divide($b, $scale);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function divideProvider(): Iterator
+    {
+        yield ['1', '2', 0, '1'];
+        yield ['1', '2', 1, '0.5'];
+        yield ['1', '2', 2, '0.50'];
+        yield ['1', '3', 3, '0.333'];
+        yield ['2', '3', 3, '0.667'];
```

Comment:
Here it is more obvious that our `Money` divides with rounding. Why do we have this behavior? Why we don't have division with truncation?

Reply:
To minimalize the error. Also, I think it's closer to our current usage (eg. form type `MoneyType` round when `scale` is defined).

Reply:
Thanks for the response, that was what I needed. Please add this reason also to documentation :-)